src/main/java/com/laytonsmith/core/AliasCore.java

```diff
@@ -215,127 +199,123 @@ public final void reload(MCPlayer player, String[] settings, boolean firstLoad)
 			}
 			return;
 		}
-		try {
-			// TODO: Maybe consider adding a ReloadOption for this? Probably not though, that would cause
-			// a huge headache.
-			compilerEnv = new CompilerEnvironment();
+
+		// Prefs is expected to be loaded before reload is called on startup, as it's needed earlier.
+		if(!firstLoad) {
+			try {
+				Prefs.init(fileLocations.getPreferencesFile());
+			} catch (IOException ex) {
+				Static.getLogger().log(Level.SEVERE, "Failed to reload preferences.", ex);
+				return;
+			}
+		} else {
 			if(Prefs.AllowDynamicShell()) {
 				MSLog.GetLogger().Log(MSLog.Tags.GENERAL, LogLevel.WARNING, "allow-dynamic-shell is set to true in "
-						+ CommandHelperFileLocations.getDefault().getProfilerConfigFile().getName()
+						+ fileLocations.getProfilerConfigFile().getName()
 						+ " you should set this to false, except during development.", Target.UNKNOWN);
 			}
+		}
 
-			if(parent.profiler == null || options.reloadProfiler()) {
-				parent.profiler = new Profiler(MethodScriptFileLocations.getDefault().getProfilerConfigFile());
-			}
+		MSLog.initialize(fileLocations.getConfigDirectory());
 
-			ProfilePoint extensionPreReload = parent.profiler.start("Extension PreReloadAliases call", LogLevel.VERBOSE);
+		Profiler profiler;
+		if(options.reloadProfiler()) {
 			try {
-				// Allow new-style extensions know we are about to reload aliases.
-				ExtensionManager.PreReloadAliases(options);
-			} finally {
-				extensionPreReload.stop();
+				profiler = new Profiler(fileLocations.getProfilerConfigFile());
+			} catch (IOException ex) {
+				Static.getLogger().log(Level.SEVERE, "Failed to create Profiler from config file.", ex);
+				return;
 			}
+		} else {
+			// If not reloading profiler, keep the old one.
+			profiler = staticRuntimeEnv.GetProfiler();
+		}
+
+		// Allow extensions know we are about to reload. Runs extension's onPreReloadAliases().
+		ProfilePoint extensionPreReload = profiler.start("Extension PreReloadAliases call", LogLevel.VERBOSE);
+		try {
+			ExtensionManager.PreReloadAliases(options);
+		} finally {
+			extensionPreReload.stop();
+		}
 
-			ProfilePoint shutdownHooks = parent.profiler.start("Shutdown hooks call", LogLevel.VERBOSE);
+		Profiles profiles;
+		try {
+			profiles = new ProfilesImpl(fileLocations.getProfilesFile());
+		} catch (IOException | Profiles.InvalidProfileException ex) {
+			MSLog.GetLogger().e(MSLog.Tags.GENERAL, ex.getMessage(), Target.UNKNOWN);
+			return;
```

Comment:
Hmm, what will this do on first load? Just prevent CH from loading up? Would it make sense to do something else here?